projects/python-client/src/datapane/client/api/ipython_utils.py

```diff
@@ -19,10 +25,219 @@ def block_to_iframe(block: BaseElement) -> str:
     Returns:
         HTML string for the block
     """
-    # importing within function scope to avoid circular dependency
     from .report.core import App
 
     app = App(block)
     block_html_string = app.stringify(template_name="ipython_template.html")
 
     return block_html_string
+
+
+@capture_event("Get Jupyter Notebook JSON")
+def get_jupyter_notebook_json() -> dict:
+    """Get the JSON for the current Jupyter notebook
+
+    Returns:
+        Notebook JSON
+    """
+    import ipynbname
+
+    try:
+        nb_path = ipynbname.path()
+        notebook_json = json.loads(open(nb_path, encoding="utf-8").read())
+    except FileNotFoundError as e:
+        raise DPError("Notebook not found. This command must be executed from within a notebook environment.") from e
+
+    return notebook_json
+
+
+@capture_event("Get VSCode Notebook JSON")
+def get_vscode_notebook_json() -> dict:
+    """Get the JSON for the current VSCode notebook
+
+    Returns:
+        Notebook JSON
+    """
+    import ipynbname
+
+    try:
+        nb_path = str(ipynbname.path())
+
+        # VSCode path name in sesssion is suffixed with -jvsc-[identifier]
+        nb_path = nb_path.split("-jvsc-")[0] + ".ipynb"
+
+        notebook_json = json.loads(open(nb_path, encoding="utf-8").read())
```

Comment:
as above - perhaps worth creating a `read_json(path: str) -> JSON` helper function at the top-level

Reply:
Done